code/game/objects/structures/props.dm

```diff
@@ -892,14 +892,44 @@
 	health = 30
 	var/inscription
 	var/obj/item/helmet
-
+	var/tagged = FALSE
+	var/engraved = FALSE
+	var/dogtag_name
+	var/dogtag_blood
+	var/dogtag_assign
 /obj/structure/prop/wooden_cross/Destroy()
 	if(helmet)
 		helmet.forceMove(loc)
 		helmet = null
+	if(tagged)
+		var/obj/item/dogtag/newInfoTag = new(loc)
+		newInfoTag.fallen_names = list(dogtag_name)
+		newInfoTag.fallen_assgns = list(dogtag_assign)
+		newInfoTag.fallen_blood_types = list(dogtag_blood)
+		fallen_list_cross.Remove(dogtag_name)
 	return ..()
 
 /obj/structure/prop/wooden_cross/attackby(obj/item/W, mob/living/user)
+	if(istype(W, /obj/item/dogtag))
+		var/obj/item/dogtag/D = W
+		if(!tagged)
+			tagged = TRUE
+			user.visible_message(SPAN_NOTICE("[user] drapes the [W] around the cross."))
+			dogtag_name = D.fallen_names[1]
+			dogtag_assign = D.fallen_assgns[1]
+			dogtag_blood = D.fallen_blood_types[1]
+			fallen_list_cross.Add(dogtag_name)
+			update_icon()
+			D.fallen_blood_types.Remove(dogtag_blood)
+			D.fallen_assgns.Remove(dogtag_assign)
+			D.fallen_names.Remove(dogtag_name)
```

Comment:
```suggestion
			D.fallen_blood_types -= D.fallen_blood_types[1]
			D.fallen_assgns -= D.fallen_assigns[1]
			D.fallen_names -= D.fallen_blood_types[1]
```
alternatively you could simplify the whole section like

```
dogtag_name = popleft(D.fallen_names)
dogtag_assign = popleft(D.fallen_assgns)
dogtag_blood = popleft(D.fallen_blood_types)
```
as this will assign and remove the first entry in the list